### PR TITLE
fix map access in (old) rule engine

### PR DESCRIPTION
**Pull request by sjka**
The object types simply did not match!

Signed-off-by: Simon Kaufmann <simon.kfm@googlemail.com>

### CODE REVIEW

File: bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java [hunk 1]
```diff
@@ -574,7 +574,9 @@ public void removeRule(TriggerTypes type, Rule rule) {
                 }
                 break;
             case TRIGGER:
-                triggerEventTriggeredRules.remove(rule);
+                for (Set<Rule> rules : triggerEventTriggeredRules.values()) {
+                    rules.remove(rule);
```
Comment (triller-telekom) [inline bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java]: Are you sure that you can remove an item from this set while you are iterating over it?

Comment (maggu2810) [inline bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java]: I don't see any problem here. He iterates over a collection of set of rules and removed something from the rules.
He does not remove something from the collection he iterates.

Comment (sjka) [inline bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java]: RTFJD:

> Returns a Collection view of the values contained in this map. The collection is backed by the map, so changes to the map are reflected in the collection, and vice-versa. If the map is modified while an iteration over the collection is in progress (except through the iterator's own remove operation), the results of the iteration are undefined. The collection supports element removal, which removes the corresponding mapping from the map, via the Iterator.remove, Collection.remove, removeAll, retainAll and clear operations. It does not support the add or addAll operations.

Comment (sjka) [inline bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java]: So I _could_ remove something from the map I iterate over, if I wanted. But I don't, as @maggu2810 correctly pointed out.

Comment (triller-telekom) [inline bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java]: Sorry my mistake, you are right.

**Pull request merged**